### PR TITLE
Make funnels query bar horizontal again

### PR DESCRIPTION
**Pull request by alexkim205**
## Changes

Real estate is hard to come by in our funnels query tab, and especially with all the new functionality that we're adding with event exclusions, conversion window, step ranges, etc. it makes sense to standardize our query across the app. 

There are a few PR's in flight (#5607 and #5704) that I want to get in before merging this one, to avoid merge conflicts. These two PR's will add event exclusions and step orderings under the OPTIONS heading.

Edit: merged #5607, and updated with exclusion filter in pic below

**Before**

Screen Shot 2021-08-24 at 4 39 02 PM

**After**

Screen Shot 2021-08-24 at 4 57 28 PM

**Mobile**

Screen Shot 2021-08-24 at 5 01 17 PM

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

**Comment by mariusandra** [comment]
Am I the only one who likes the current vertical layout? Without scrolling, I can configure a funnel and see the results right next to it:
image

Not directly opposed to this, but... 🤔

**Review by Twixes — COMMENTED** [review]
Code looks good to me, and I see the point in unifying this layout across all insights (instead of funnels being the odd one out).

I noticed one UX issue, in that the saved funnels pane takes up a needlessly large amount of space on screens any wider than a 13" MacBook's, let alone a 21:9 display.
Screen Shot 2021-08-25 at 11 54 54
I think we should be able to easily fit that to the right of the query pane.

Regarding final UI feedback, I'd really like to hear from @clarkus though ;)

**Comment by alexkim205** [comment]
> Am I the only one who likes the current vertical layout? Without scrolling, I can configure a funnel and see the results right next to it

There's definitely value in this vertical layout, but I think the same can be said about the other insight graphs. An added advantage to the horizontal layout would be the more options we have, the user will have to scroll down further which makes it harder to discover these features. With @Twixes's suggestion, the tab will take up less space too. @clarkus what do you think?

**Comment by clarkus** [comment]
The insights page layout is something I've been thinking about a lot. I do think it would be good to have some consistency across insight types, but I also agree with @mariusandra that it's valuable to see both the query builder and the visualization at the same time. I think I am mostly torn on the timing of these changes. I'd be in favor of it if it allowed us to include other query builder options like separate "add" and "duplication" actions, or renaming query steps to use friendlier values. 

The "funnels saved to this project" card should probably be removed once we have #3408 wrapped up. 

Just a quick note that our target viewport size for most users is 1280px wide by 720px tall. This is a rough goal, but it does give us some baseline to evaluate layout changes.

**Comment by alexkim205** [comment]
Thanks for the context, that makes sense. I would argue that this horizontal layout **would** give us more space to add other query builder options. It's difficult to have add + duplicate + remove + rename/edit icons inline with an action filter in the current vertical layout, and forces us to account for wrapping edge cases that is otherwise inconsistent with our other queries (like we saw w/ exclusion events). I think this layout would actually unlock a lot of functionality that we can implement in funnel queries. 

However I wonder if there's a way to have the best of both worlds without sacrificing the side-by-side layout.

Got it, I'll add a reminder to myself to remove "funnels saved to this project" once #3408 is wrapped up.

**Comment by clarkus** [comment]
> However I wonder if there's a way to have the best of both worlds without sacrificing the side-by-side layout.

This is kind of where I was going with the idea at https://github.com/PostHog/posthog/issues/5141. I don't know if it's the right solution, but something that balances the scale of building a query but allows the visualization to have as much space as necessary to effectively communicate everything. It's tricky.

**Comment by alexkim205** [comment]
I think #5141 is where we want to be and could be taken up under future query revamp work. 

Given the feedback here and the multiple things we want to add to the funnels display (including this #5704), I think there's a strong usability argument for changing the query experience in funnels to be horizontal.

**Comment by alexkim205** [comment]
@Twixes rerequesting your review after moving saved funnels up.

Screen Shot 2021-08-25 at 1 25 38 PM

**Comment by mariusandra** [comment]
I'm not opposed to this layout change, but as someone who regularly uses a large/wide screen, I feel this will give me a reason to scroll when there's enough space available. So could we be responsive here, and show it as a sidebar for large screens, and on top if the area isn't large enough to meaningfully show enough funnel steps?

Also, should this be feature flagged... or yolo? :)

LGTM for the code otherwise.

**Comment by clarkus** [comment]
> I'm not opposed to this layout change, but as someone who regularly uses a large/wide screen, I feel this will give me a reason to scroll when there's enough space available. So could we be responsive here, and show it as a sidebar for large screens, and on top if the area isn't large enough to meaningfully show enough funnel steps?

👍 for updated breakpoints. I have been working on this problem a bit as I work on other issues, and I don't have a solution yet. I will spend some time on this soon to see if we can get to a shorter-term layout change that makes everyone happy. :)

**Comment by alexkim205** [comment]
For sure, I think the sidebar for our large screens still makes sense. I'll also feature flag this CC @paolodamico for context

**Comment by alexkim205** [comment]
With feature flag


https://user-images.githubusercontent.com/13460330/131008864-3b292c82-9998-4e5f-9cce-d94e0f006311.mov



Without feature flag



https://user-images.githubusercontent.com/13460330/131008880-66387389-5e77-4951-984f-656c4afd0b87.mov

**Comment by paolodamico** [comment]
Thanks for feature flagging this @alexkim205! Just managing to catch up with everything here. Personally I strongly prefer the vertical layout even if it's inconsistent with the rest of the insights, it's what makes the most sense in this particular context. Imagine for instance having 5+ steps, it'll look quite cumbersome. 

There's definitely value in having a consistent layout, I just don't think this is the answer. **My proposal would be to keep this offline and instead continue on @clarkus's proposal** (#5141) to figure out the layout that makes the most sense for all use cases.

A final note is that we need to be conscious if we're going to put more effort into redoing the insights builder, it'll be a ton of work on Design, Product & Engineering. Is it the right time to focus on this?

**Comment by alexkim205** [comment]
Thanks for the input @paolodamico. We can definitely revisit this later down the road, and have it live behind a feature flag for now until we figure out a more consistent layout. Let's revisit this when we have more pieces to the puzzle here

File: frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelTab.tsx [hunk 1]
```diff
@@ -27,107 +44,197 @@ export function FunnelTab(): JSX.Element {
         closeModal()
     }
 
+    console.log('Small screen', isSmallScreen)
```
[the remark this answers is not in the document]
Comment (alexkim205) [reply frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelTab.tsx:47]: ah noticed this almost immediately, and nixed it in my other PR that's merged in.

**Pull request merged**